### PR TITLE
feat: ArgoCD plugin: add environment variable KUBEVERSION and KUBE_API_VERS…

### PR DESCRIPTION
**Pull request by mayzhang2000**
…IONS.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Comment by mayzhang2000** [comment]
Please do not merge yet. I want to get initial review first. After that I also need add additional verification for env variable KUBE_API_VERSION.

File: reposerver/repository/repository.go [hunk 1]
```diff
@@ -633,6 +633,8 @@ func runConfigManagementPlugin(appPath string, envVars *v1alpha1.Env, q *apiclie
 		env = append(env, environ...)
 	}
 	env = append(env, q.ApplicationSource.Plugin.Env.Environ()...)
+	env = append(env, "KUBEVERSION="+q.KubeVersion)
```
Comment (alexmt) [inline reposerver/repository/repository.go]: To be consistent please use `KUBE_VERSION` instead of `KUBEVERSION`

**Review by alexmt — CHANGES_REQUESTED** [review]

File: reposerver/repository/repository.go (hunk 1 above)
Comment (mayzhang2000) [reply reposerver/repository/repository.go]: :+1: Fixed. Also added test verification of KUBE_API_VERSION.

File: test/e2e/custom_tool_test.go [hunk 2]
```diff
@@ -136,5 +138,42 @@ func TestCustomToolWithEnv(t *testing.T) {
...
+			expectedApiVersion := GetApiVersions()
+			expectedApiVersionSlice := strings.Split(expectedApiVersion, ",")
+			sort.Strings(expectedApiVersionSlice)
+
+			output, err := Run("", "kubectl", "-n", DeploymentNamespace(), "get", "cm", Name(), "-o", "jsonpath={.metadata.annotations.KubeApiVersion}")
+			assert.NoError(t, err)
+			outputSlice := strings.Split(output, ",")
+			sort.Strings(outputSlice)
+
+			assert.True(t, testEq(expectedApiVersionSlice, outputSlice))
```
Comment (alexmt) [inline test/e2e/custom_tool_test.go]:
```suggestion
			assert.EqualValues(t, expectedApiVersionSlice, outputSlice)
```

File: test/e2e/custom_tool_test.go [hunk 3]
```diff
@@ -136,5 +138,42 @@ func TestCustomToolWithEnv(t *testing.T) {
...
 		})
 }
+
+func testEq(a, b []string) bool {
```
Comment (alexmt) [inline test/e2e/custom_tool_test.go]: Golang built-in testing library has a similar method already: `assert.EqualValues`. Please use it instead.

**Review by alexmt — CHANGES_REQUESTED** [review]

**Review by alexmt — APPROVED** [review]
LGTM!

**Comment by jannfis** [comment]
Should this maybe be updated in the documentation as well, i.e. https://argoproj.github.io/argo-cd/user-guide/build-environment/

**Pull request merged**